posthog/models/event.py

```diff
@@ -223,17 +224,24 @@ def query_retention(self, filters, team, start_entity: Optional[Entity] = None)
 
         filtered_events = events.filter(filters.date_filter_Q).filter(filters.properties_to_Q(team_id=team.pk))
 
-        first_date = (
-            filtered_events.annotate(first_date=TruncDay("timestamp")).values("first_date", "person_id").distinct()
-        )
+        def _determineTrunc(subject: str, period: str) -> Union[TruncHour, TruncDay, TruncWeek, TruncMonth]:
```

Comment:
None of the code paths return `TruncMonth`, this seems actively misleading to include in the `Union`.

Consider also using the `raise ValueError` pattern for the default here as well.